tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -29,521 +29,514 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
+						Skip("This is a OpenShift specific scenario, skipping")
+					}
+
+					stdOut := helper.CmdShouldFail("odo", "link", "EtcdCluster")
+					Expect(stdOut).To(ContainSubstring("invalid service name"))
 
-	Context("When using from-file option", func() {
+					stdOut = helper.CmdShouldFail("odo", "link", "EtcdCluster/")
+					Expect(stdOut).To(ContainSubstring("invalid service name"))
 
-		var tmpContext string
+					stdOut = helper.CmdShouldFail("odo", "link", "/example")
+					Expect(stdOut).To(ContainSubstring("invalid service name"))
+				})
 
-		JustBeforeEach(func() {
-			tmpContext = helper.CreateNewContext()
-			preSetup()
-		})
+				When("odo push is executed", func() {
+					JustBeforeEach(func() {
+						helper.CmdShouldPass("odo", "push")
+					})
 
-		JustAfterEach(func() {
-			helper.DeleteDir(tmpContext)
-			cleanPreSetup()
-		})
+					JustAfterEach(func() {
+					})
+					It("should fail if the provided service doesn't exist in the namespace", func() {
+						if os.Getenv("KUBERNETES") == "true" {
+							Skip("This is a OpenShift specific scenario, skipping")
+						}
+						stdOut := helper.CmdShouldFail("odo", "link", "EtcdCluster/example")
+						Expect(stdOut).To(ContainSubstring("Couldn't find service named %q", "EtcdCluster/example"))
+					})
+				})
+
+				When("an EtcdCluster instance is created in dryRun mode", func() {
+
+					var stdOut string
+
+					JustBeforeEach(func() {
+						stdOut = helper.CmdShouldPass("odo", "service", "create", fmt.Sprintf("%s/EtcdCluster", etcdOperator), "--dry-run", "--project", commonVar.Project)
+					})
+
+					JustAfterEach(func() {
+					})
+
+					It("should only output the definition of the CR that will be used to start service", func() {
+						helper.MatchAllInOutput(stdOut, []string{"apiVersion", "kind"})
+					})
+
+					When("the output of the command is stored in a file", func() {
+
+						var fileName string
+
+						JustBeforeEach(func() {
+							randomFileName := helper.RandString(6) + ".yaml"
+							fileName = filepath.Join(os.TempDir(), randomFileName)
+							if err := ioutil.WriteFile(fileName, []byte(stdOut), 0644); err != nil {
+								fmt.Printf("Could not write yaml spec to file %s because of the error %v", fileName, err.Error())
+							}
+						})
+
+						JustAfterEach(func() {
+							// TODO delete file
+						})
+
+						When("a service is created from the output of the dryRun command with no name", func() {
+							JustBeforeEach(func() {
+								helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName, "--project", commonVar.Project)
+							})
+
+							JustAfterEach(func() {
+								helper.CmdShouldPass("odo", "service", "delete", "EtcdCluster/example", "-f")
+							})
+
+							When("odo push is executed", func() {
+								JustBeforeEach(func() {
+									helper.CmdShouldPass("odo", "push")
+								})
+
+								JustAfterEach(func() {
+								})
+
+								It("should create pods in running state", func() {
```

Comment:
This particular spec is repeating at a few places and the code looks the same in all instances. Can we create a function for the repeating code and use it instead?

Reply:
Yes, it would be nice to reuse some code with functions. I will look and see where it would be necessary

Reply:
done